test/framework/src/main/java/org/opensearch/test/NodeConfigurationSource.java

```diff
@@ -63,4 +64,9 @@ public Path nodeConfigPath(int nodeOrdinal) {
     public Collection<Class<? extends Plugin>> nodePlugins() {
         return Collections.emptyList();
     }
+
+    /** Returns a map corresponding to plugin dependencies to other classpath plugins */
+    public Map<Class<? extends Plugin>, Class<? extends Plugin>> extendedPlugins() {
```

Comment:
I think this is the last place where we need `PluginInfo` (from classpath): `public Class<PluginInfo> classpathPlugins() {`